src/lua/init.lua

```diff
@@ -57,27 +57,28 @@ local function get_require_overload_count(stack)
     return 0
 end
 
--- Make path relative to the current directory.
-local function remove_root_directory(path)
-    local pwd = os.getenv("PWD")
-    if pwd == nil then
-        return path
+-- Return longest common subpath of two paths.
+local function get_common_subpath(path1, path2)
+    local result = '/'
+    for _, dir in ipairs(path1:split('/')) do
+        local common = fio.pathjoin(result, dir)
+        if path2:startswith(common) then
```

Comment:
This doesn't look correct. For example, after the following change `per_module_log_unit_test` still passes although it shouldn't:

<details><summary>diff</summary>

```diff
diff --git a/test/app-luatest/per_module_log_unit_test.lua b/test/app-luatest/per_module_log_unit_test.lua
index b87e5992c05d..261cb8a4e4d9 100644
--- a/test/app-luatest/per_module_log_unit_test.lua
+++ b/test/app-luatest/per_module_log_unit_test.lua
@@ -8,7 +8,7 @@ local module_name_from_filename = tarantool._internal.module_name_from_filename
 g.test_strip_cwd_from_path = function()
     local testcases = {
         { cwd = '/', path = '/aaa/bbb/ccc', expected = 'aaa/bbb/ccc' },
-        { cwd = '/home/aaa/bbb', path = '/home/aaa/bbb/ccc/ddd', expected = 'ccc/ddd' },
+        { cwd = '/home/aaa/bbb', path = '/home/aaa/bbbccc/ddd', expected = 'ccc/ddd' },
         { cwd = '/home/some/other/path', path = '/home/aaa/bbb', expected = 'aaa/bbb' },
         { cwd = '/totally/different', path = '/home/aaa/bbb', expected = 'home/aaa/bbb' },
         { cwd = '/doesnt/matter', path = 'aaa/bbb', expected = 'aaa/bbb' },
```

</details>

Anyway, I don't quite get why you split the path by `/`. Why not simply find the longest common sub-string?

Reply:
> This doesn't look correct. For example, after the following change `per_module_log_unit_test` still passes although it shouldn't:
 
Thanks, fixed.

> Anyway, I don't quite get why you split the path by `/`. Why not simply find the longest common sub-string?

With `split` it's easier to handle situations like this:
```lua
{ cwd = '/home/common_qwe', path = '/home/common_asd/aaa', expected = 'common_asd/aaa' }
```